src/connectors/source.rs

```diff
@@ -834,21 +840,24 @@ where
     async fn handle_link(
         &mut self,
         port: Cow<'static, str>,
+        tx: Sender<Result<()>>,
         mut pipelines: Vec<(DeployEndpoint, pipeline::Addr)>,
-    ) -> Control {
+    ) -> Result<Control> {
         let pipes = if port.eq_ignore_ascii_case(OUT.as_ref()) {
             &mut self.pipelines_out
         } else if port.eq_ignore_ascii_case(ERR.as_ref()) {
             &mut self.pipelines_err
         } else {
             error!("{} Tried to connect to invalid port: {}", &self.ctx, &port);
-            return Control::Continue;
+            return Ok(Control::Continue);
         };
         // We can not move this to the system flow since we need to know about transactionality
         for (pipeline_url, p) in &pipelines {
             self.ctx.swallow_err(
                 p.send_mgmt(pipeline::MgmtMsg::ConnectInput {
                     endpoint: DeployEndpoint::new(&self.ctx.alias, &port, pipeline_url.meta()),
+                    port: Cow::from(pipeline_url.port().to_string()),
+                    tx: tx.clone(),
```

Comment:
Each `ConnectInput` will send a response to this channel, which expects only 1 message when the whole outer operation is done. This will lead to the link reported as ready when in fact it might not be ready yet.

I'd suggest to create a `bounded(1)` channel here and wait for as many messages as we have pipelines here.
If a send failed, we might want to simply return the `Err`.